### ISSUE TITLE
Handing DRBD during new update tool

### ISSUE THREAD

**Issue by digimer** [ClusterLabs/anvil#349]
14:34 <fabbione> [root@mk-a07n01 log]# rpm -qa|grep drbd
14:34 <fabbione> akmod-drbd-9.2.4-1.el8.x86_64
14:34 <fabbione> kmod-drbd-4.18.0-448.el8-9.2.2-1.el8.x86_64
14:34 <fabbione> kmod-drbd-9.2.4-1.el8.x86_64
14:34 <fabbione> kmod-drbd-4.18.0-408.el8-9.2.1-1.el8.x86_64
14:34 <fabbione> kmod-drbd-4.18.0-497.el8-9.2.4-1.el8.x86_64
14:34 <fabbione> kmod-drbd-4.18.0-489.el8-9.2.3-1.el8.x86_64
14:34 <fabbione> drbd90-utils-9.24.0-1.el8.x86_64
14:36 <fabbione> 20:36 <fabbione> [root@mk-a07n01 log]# rpm -qa|grep drbd
14:36 <fabbione> 20:36 <fabbione> akmod-drbd-9.2.4-1.el8.x86_64
14:36 <fabbione> 20:36 <fabbione> kmod-drbd-9.2.4-1.el8.x86_64
14:36 <fabbione> 20:36 <fabbione> drbd90-utils-9.24.0-1.el8.x86_64
14:38 <fabbione> akmod-drbd-9.2.4-1.el8.x86_64 <- drbd version
14:38 <fabbione> remove all of kmod-drbd | grep -v kmod-drbd-$(drbd version)

**Comment by digimer** [comment]
14:40 <fabbione> [root@mk-a07dr02 yum.repos.d]# rpm -qa| grep kmod
14:40 <fabbione> kmod-drbd-4.18.0-497.el8-9.2.4-1.el8.x86_64
14:40 <fabbione> akmods-0.5.7-8.el8.noarch
14:40 <fabbione> kmodtool-1.1-5.el8.noarch
14:40 <fabbione> kmod-drbd-9.2.4-1.el8.x86_64
14:40 <fabbione> kmod-25-19.el8.x86_64
14:40 <fabbione> kmod-drbd-4.18.0-408.el8-9.2.1-1.el8.x86_64
14:40 <fabbione> kmod-drbd-4.18.0-448.el8-9.2.2-1.el8.x86_64
14:40 <fabbione> akmod-drbd-9.2.4-1.el8.x86_64
14:40 <fabbione> kmod-libs-25-19.el8.x86_64

**Comment by digimer** [comment]
00:16 < fabbione> the error / issue shows when there is a drbd update (kmod) and no kernel update
00:17 < fabbione> to trigger that problem, you need to downgrade drbd-kmod in your test env
00:17 < fabbione> and make sure it needs to be upgraded
00:17 < fabbione> best if there is no kernel update involved
00:17 < fabbione> akmod or yum will fail to install the new kmod

**Comment by digimer** [comment]
reproduced

**Comment by digimer** [comment]
This should be resolved in PR#362

**Comment by fabbione** [comment]
the code in PR#362 is too invasive and would remove all kmod-drbd. This would cause anvil-* to also be removed.

```
Taking as example:

# rpm -qa | grep drbd
drbd90-utils-9.24.0-1.el8.x86_64
kmod-drbd-4.18.0-500.el8-9.2.4-1.el8.x86_64
kmod-drbd-9.2.4-1.el8.x86_64
akmod-drbd-9.2.4-1.el8.x86_64

we need to remove:
kmod-drbd-4.18.0-500.el8-9.2.4-1.el8.x86_64

The logic must be applied _before_ any yum update and it goes:

determine current DRBD version:
# rpm -q akmod-drbd --queryformat="%{version}-%{release}\n"
9.2.4-1.el8

determine list of currently installed kmod in that version for kernels:
# rpm -qa | grep "kmod-drbd.*el.*-9.2.4-1.el8.*"
kmod-drbd-4.18.0-500.el8-9.2.4-1.el8.x86_64

Another alternative is to use:

find /lib/modules/ -name "*drbd*" -exec rpm -qf {} \; |grep -v "not owned" | sort -u

```

The modules _must_ be removed before the yum update or the akmod version will not match and false the results of the removal.

Otherwise you can upgrade first, get the error, then remove using method 2 (with find) and then re-run the update. IMHO running 2 updates is more expensive than cleaning and then updating.

**Comment by digimer** [comment]
Recalling our discussion on IRC, the regex filters only the built kmod RPMs, and won't pull out the core RPM that would have pulled out anvil-X rpms. Once confirmed this is the case, please close. If not, let me know.

**Comment by digimer** [comment]
Resolved in commit f998a44.

**Comment by fabbione** [comment]
It seems to be doing the right thing now. I will test it a couple of more times.

**Comment by fabbione** [comment]
This is fixed in pr #376
